### DISCUSSION TITLE
Table with a large number of rows

### DISCUSSION THREAD

**Discussion by kfmfe04** [Textualize/textual#379] [Q&A]
Suppose I have a large Table with 120 rows and 2 columns, but the screen currently is 80 rows.

Using traditional ncurses, I would break the Table into two tables, side-by-side, 80 and 40.
Under the textual approach, looking at the screen dimensions is doesn't seem right.

I'm a bit perplexed because I want the result to play well with the automatic formatting in textual.
But the only approach I can come up with involves peeking at screen height.

Would it be ok to peek at screen height?  If not, can you suggest a better approach?

**Comment by willmcgugan** [comment]
Have you seen the `big_table.py` example?

**Reply by kfmfe04** [reply]
Yes, I have, but I've completely forgotten about that until you mentioned it.

As impressive as it looks, I'm trying to avoid scrollbars as much as possible.
If the Table gets that big, I'll create a 2nd page/view instead, for minimal keyboard interaction.
But thanks for the reminder - that shows me I can (and should) keep it in one table, until I get to page 2.

Only thing left to mod is making the number of rows depend on screen dimensions, or more specifically, the size of the container I'm in (maybe Pane)!

Any hints on where I should look?  (As self.height and self.width are being set to None for auto-sizing, it's not obvious).  In some rich code, I see child_width and child_height being calculated on-the-fly, but it's not stored.

TY.
